server/middlewares/validations.js

```diff
@@ -0,0 +1,77 @@
+const validateUsername = (username) => {
+  const usernamePattern = /^[a-zA-Z]+[a-zA-Z0-9_]+$/;
+  return usernamePattern.test(username);
+};
+
+const validateEmail = (email) => {
+  const emailPattern = /^[a-zA-Z0-9]+[-_.]*[a-zA-Z0-9]*@[a-zA-Z0-9]+\.(?:[a-zA-Z0-9]+)*$/;
+  return emailPattern.test(email);
+};
+
+const validatePassword = (password) => {
+  const passwordPattern = /^[0-9a-z]+$/;
```

Comment:
how does this regex work?

Reply:
The ticket specified that an example of an invalid password is one which is not alpha-numeric. I see an omission though. The regex should be /^[0-9a-zA-Z]+$/ to give room for capital alphabets